components/product/VariantSelector.tsx

```diff
@@ -17,40 +16,47 @@ export const VariantSelector = ({
   selectedVariantID,
 }: VariantSelectorProps) => {
   const paths = usePaths();
+  const router = useRouter();
+
+  const [value, setValue] = useState(selectedVariantID);
 
   const variants = product.variants;
+
+  // Skip displaying selector when theres less than 2 variants
   if (!variants || variants.length === 1) {
     return null;
   }
+
+  const onChange = (event: React.FormEvent<HTMLSelectElement>) => {
+    const target = event.target as HTMLSelectElement;
```

Comment:
```suggestion
    const target = event.target;
```
This type should be auto-inferred based on the fn argument, are you sure it's the right type?

Reply:
StackOverflow for the rescue - changed to `React.ChangeEvent<HTMLSelectElement>` which seems to be the right one.